nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/src/main/java/org/apache/nifi/flowanalysis/rules/RestrictBackpressureSettings.java

```diff
@@ -0,0 +1,242 @@
+                    if (Double.compare(sizeThreshold, maxSize) > 0) {
+                        results.add(buildViolation(connection,
+                                idComponent.get(connection.getSource().getId()),
+                                idComponent.get(connection.getDestination().getId()),
+                                BackpressureViolationType.BP_SIZE_THRESHOLD_ABOVE_LIMIT,
+                                getViolationMessage(BackpressureViolationType.BP_SIZE_THRESHOLD_ABOVE_LIMIT, connection.getBackPressureDataSizeThreshold(), context.getProperty(SIZE_MAX).getValue())));
+                    }
+                }
+            );
+
+        return results;
+    }
+
+    private GroupAnalysisResult buildViolation(final VersionedConnection connection, final VersionedComponent source,
+            final VersionedComponent destination, final BackpressureViolationType backpressureViolationType, final String violationMessage) {
+        if (!(source instanceof VersionedProcessor) && !(destination instanceof VersionedProcessor)) {
+            // connection between two components that are not processors and cannot be invalid, setting violation on connection
+            return GroupAnalysisResult.forComponent(connection,
+                    connection.getIdentifier() + "_" + backpressureViolationType.getId(),
+                    getLocationMessage(connection, source, destination) + violationMessage).build();
+        } else if (source instanceof VersionedProcessor) {
+            // defining violation on source processor
+            return GroupAnalysisResult.forComponent(source,
+                    connection.getIdentifier() + "_" + backpressureViolationType.getId(),
+                    getLocationMessage(connection, source, destination) + violationMessage).build();
+        } else {
+            // defining violation on destination processor
+            return GroupAnalysisResult.forComponent(destination,
+                    connection.getIdentifier() + "_" + backpressureViolationType.getId(),
+                    getLocationMessage(connection, source, destination) + violationMessage).build();
+        }
+    }
+
+    private String getLocationMessage(final VersionedConnection connection, final VersionedComponent source, final VersionedComponent destination) {
+        if (source == null || destination == null) {
+            return "The connection [" + connection.getIdentifier() + "] is violating the rule for backpressure settings. ";
+        }
+        return "The connection [" + connection.getIdentifier() + "] connecting " + source.getName() + " [" + source.getIdentifier() + "] to "
+                + destination.getName() + " [" + destination.getIdentifier() + "] is violating the rule for backpressure settings. ";
+    }
+
+    private String getViolationMessage(final BackpressureViolationType backpressureViolationType, final String configured, final String limit) {
+        switch (backpressureViolationType) {
```

Comment:
This switch statement looks like a candidate for using enhanced switch and removing the default case that isn't needed for switching over an Enum.
```
return switch (backpressureViolationType) {
  case BP_COUNT_THRESHOLD_ABOVE_LIMIT ->
    "blah";
  case BP_COUNT_THRESHOLD_BELOW_LIMIT ->
    "blah";
  case BP_SIZE_THRESHOLD_ABOVE_LIMIT ->
    "blah";
  case BP_SIZE_THRESHOLD_BELOW_LIMIT ->
    "blah";
};
```